File: android_mobile/src/main/java/com/alexstyl/specialdates/ui/ViewFader.java
package com.alexstyl.specialdates.ui;

import android.view.ViewGroup;

import static android.view.View.GONE;
import static android.view.View.VISIBLE;

public final class ViewFader {

    public void hideContentOf(ViewGroup viewGroup) {
        for (int i = 0; i < viewGroup.getChildCount(); i++) {
            viewGroup.getChildAt(i).setVisibility(GONE);
        }
    }

    public void showContent(ViewGroup viewGroup) {
        for (int i = 0; i < viewGroup.getChildCount(); i++) {
            viewGroup.getChildAt(i).setVisibility(VISIBLE);
        }
    }

}

File: android_mobile/src/main/java/com/alexstyl/specialdates/ui/ViewFader.kt
package com.alexstyl.specialdates.ui

import android.view.ViewGroup

import android.view.View.GONE
import android.view.View.VISIBLE

object ViewFader {

    fun hideContentOf(viewGroup: ViewGroup) {
        for (i in 0 until viewGroup.childCount) {
            viewGroup.getChildAt(i).visibility = GONE
        }
    }

    fun showContent(viewGroup: ViewGroup) {
        for (i in 0 until viewGroup.childCount) {
            viewGroup.getChildAt(i).visibility = VISIBLE
        }
    }

}
